CMSgov/mint-app
# [MINT-2650]/model by solution bug fix

# MINT-2650

## Description
Duplicate model plans were being returned to the front end for models that use a solution multiple times. To fix this the type needed to be updated.

As part of the work, a data loader for the operational solution by key was needed. Discussing with the front end team, it was determined we should return the possible operational solution instead of multiple solution instances.

Note, this dataloader has deviated somewhat from previous data loader implementations. This is to facilitate the eventual migration of the dataloaders to a different library.



## How to test this change
1. Bring up MINT.
2. Make a model plan
3. For multiple model plans, select the same solution for multiple needs.
4. For your homepage customization, ensure you only see one instance of that model plan 

(you can also achieve this by logging in as MINT, seeding the database, and updating a model plan that is already using 4 innovation to use it again).

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
